cmd/fix/feegrant/allowance.go

```diff
@@ -0,0 +1,113 @@
+
+	"github.com/forbole/bdjuno/v2/database"
+
+	"sort"
+
+	tmctypes "github.com/tendermint/tendermint/rpc/core/types"
+
+	feegranttypes "github.com/cosmos/cosmos-sdk/x/feegrant"
+	"github.com/rs/zerolog/log"
+)
+
+// allowanceCmd returns the Cobra command allowing to fix all things related to fee grant allowance
+func allowanceCmd(parseConfig *parse.Config) *cobra.Command {
+	return &cobra.Command{
+		Use:   "allowance",
+		Short: "Fix granted and revoked allowances to the latest height",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			parseCtx, err := parse.GetParsingContext(parseConfig)
+			if err != nil {
+				return err
+			}
+
+			// Get the database
+			db := database.Cast(parseCtx.Database)
+
+			// Build feegrant module
+			feegrantModule := feegrant.NewModule(parseCtx.EncodingConfig.Marshaler, db)
+
+			// Get the accounts
+			addresses, err := db.GetAccounts()
+			if err != nil {
+				return err
+			}
+
+			for _, address := range addresses {
+				// Collect all the transactions
+				var txs []*tmctypes.ResultTx
+
+				// Get all the MsgGrantAllowance txs
+				query := fmt.Sprintf("set_feegrant.granter='%s'", address)
+				grantAllowanceTxs, err := utils.QueryTxs(parseCtx.Node, query)
+				if err != nil {
+					return err
+				}
+				txs = append(txs, grantAllowanceTxs...)
+
+				query = fmt.Sprintf("set_feegrant.grantee='%s'", address)
+				grantAllowanceTxs, err = utils.QueryTxs(parseCtx.Node, query)
+				if err != nil {
+					return err
+				}
+				txs = append(txs, grantAllowanceTxs...)
+
+				// Get all the MsgRevokeAllowance txs
+				query = fmt.Sprintf("revoke_feegrant.granter='%s'", address)
+				revokeAllowanceTxs, err := utils.QueryTxs(parseCtx.Node, query)
+				if err != nil {
+					return err
+				}
+				txs = append(txs, revokeAllowanceTxs...)
+
+				query = fmt.Sprintf("revoke_feegrant.grantee='%s'", address)
+				revokeAllowanceTxs, err = utils.QueryTxs(parseCtx.Node, query)
+				if err != nil {
+					return err
+				}
+				txs = append(txs, revokeAllowanceTxs...)
+
+				// Sort the txs based on their ascending height
+				sort.Slice(txs, func(i, j int) bool {
+					return txs[i].Height < txs[j].Height
+				})
+
+				for _, tx := range txs {
+					log.Debug().Int64("height", tx.Height).Msg("parsing transaction")
+
+					transaction, err := parseCtx.Node.Tx(hex.EncodeToString(tx.Tx.Hash()))
+					if err != nil {
+						return err
+					}
+
+					// Handle only the MsgGrantAllowance and MsgRevokeAllowance instances
+					for index, msg := range transaction.GetMsgs() {
+						_, isMsgGrantAllowance := msg.(*feegranttypes.MsgGrantAllowance)
+						_, isMsgRevokeAllowance := msg.(*feegranttypes.MsgRevokeAllowance)
+
+						if !isMsgGrantAllowance && !isMsgRevokeAllowance {
+							continue
+						}
+
+						err = feegrantModule.HandleMsg(index, msg, transaction)
+						if err != nil {
+							return fmt.Errorf("error while handling feegrant module message: %s", err)
+						}
+					}
+				}
+			}
```

Comment:
Instead of searching for all given and revoked grants for all accounts there is a way to search for all grant and revoke grant transactions. This should allow us to both remove the `for` cycle as well as removing a lot of code here: 

```suggestion
			// Collect all the transactions
			var txs []*tmctypes.ResultTx

			// Get all the MsgGrantAllowance txs
			query := fmt.Sprintf("message.action='%s'", feegranttypes.EventTypeSetFeeGrant)
			grantAllowanceTxs, err := utils.QueryTxs(parseCtx.Node, query)
			if err != nil {
				return err
			}
			txs = append(txs, grantAllowanceTxs...)

			// Get all the MsgRevokeAllowance txs
			query = fmt.Sprintf("message.action='%s'", feegranttypes.EventTypeRevokeFeeGrant)
			revokeAllowanceTxs, err := utils.QueryTxs(parseCtx.Node, query)
			if err != nil {
				return err
			}
			txs = append(txs, revokeAllowanceTxs...)

			// Sort the txs based on their ascending height
			sort.Slice(txs, func(i, j int) bool {
				return txs[i].Height < txs[j].Height
			})

			for _, tx := range txs {
				log.Debug().Int64("height", tx.Height).Msg("parsing transaction")
				transaction, err := parseCtx.Node.Tx(hex.EncodeToString(tx.Tx.Hash()))
				if err != nil {
					return err
				}

				// Handle only the MsgGrantAllowance and MsgRevokeAllowance instances
				for index, msg := range transaction.GetMsgs() {
					_, isMsgGrantAllowance := msg.(*feegranttypes.MsgGrantAllowance)
					_, isMsgRevokeAllowance := msg.(*feegranttypes.MsgRevokeAllowance)

					if !isMsgGrantAllowance && !isMsgRevokeAllowance {
						continue
					}

					err = feegrantModule.HandleMsg(index, msg, transaction)
					if err != nil {
						return fmt.Errorf("error while handling feegrant module message: %s", err)
					}
				}
			}
		}
```